CONTRIBUTING.md

```diff
@@ -32,41 +67,67 @@ source venv/bin/activate
 
 ### Testing
 
-To run tests, run the following
+To run tests, first build the charm as described above, then run the following
 
 ```shell
 tox -e format       # update your code according to linting rules
 tox -e lint         # code style
 tox -e unit         # unit tests
-tox -m integration  # integration tests, running on juju 2.
+tox -m integration  # integration tests, running on juju 3.
 tox                 # runs 'format', 'lint', and 'unit' environments
 ```
 
-Integration tests can be run for separate areas of functionality:
+Integration tests can be run for separate files:
 
 ```shell
-tox -e charm-integration      # basic charm integration tests
-tox -e tls-integration        # TLS-specific integration tests
-tox -e client-integration     # Validates the `opensearch-client` integration
-tox -e ha-integration         # HA tests
-tox -e h-scaling-integration  # HA tests specific to horizontal scaling
+tox -e integration -- tests/integration/ha/test_storage.py
 ```
 
-If you're running tests on juju 3, run the following command to change libjuju to the correct version:
+#### Running different versions of libjuju
+
+To try different versions of libjuju, run:
 
 ```shell
-export LIBJUJU_VERSION_SPECIFIER="==3.3.0.0"
+poetry add --lock --group integration,unit juju@<YOUR CHOSEN VERSION>
 ```
 
-## Build charm
+#### Testing Backups
 
-Build the charm in this git repository using:
+Backup testing installs microceph and, optionally, can run on AWS and GCP object stores.
+For that, pass the access / secret / service account information as env. variables.
+
+To run the test only against microceph:
 
 ```shell
-charmcraft pack
+tox -e integration -- tests/integration/ha/test_backups.py --group='microceph' # test backup service
+```
+
+And against public clouds + microceph:
+
+```shell
+SECRETS_FROM_GITHUB=$(cat <path-to>/credentials.json) tox -e integration -- tests/integration/ha/test_backups.py
```

Comment:
Ack, but that is needed for us to test locally. I added more text explaining this is development only and should be done only on your local machine

Reply:
@carlcsaposs-canonical my recommendation is to always run tests locally as well. That makes our own lives easier for testing. Indeed, microceph is not installing because of that: https://pastebin.ubuntu.com/p/9JjmVQ6G9G/

Reply:
Sorry, not "to always run" but to always "have the option". In any case, this is a discussion that we can take offline.

Reply:
It would make sense to stay consistent - given that this PR is about moving to DP workflows, we should aim to have the same UX as the rest.

Reply:
Ack, going to resolve this comment. It was also due to some settings left in tox.ini. I will clean them up, if not already done.